File: 2024-2025/tecnologia_programacion/procesador_instrucciones/c++/main.cc
#include "programa1.h"
#include "programa2.h"
#include "programa3.h"
using namespace std;

int main () {
    // PRUEBA PROGRAMA 1 - SUMA DE DOS NÚMEROS
    cout << "Primer programa: SUMA DE DOS NUMEROS" << endl;
    programa1 p1;
    cout << "Programa:" << endl;
    p1.listar();
    cout << endl << "Ejecucion: " << endl;
    p1.ejecutar();
    cout << endl;

    // PRUEBA PROGRAMA 2 - CUENTA ATRÁS
    cout << "Segundo programa: CUENTA ATRAS" << endl;
    programa2 p2;
    cout << "Programa:" << endl;
    p2.listar();
    cout << endl << "Ejecucion: " << endl;
    p2.ejecutar();
    cout << endl;

    // PRUEBA PROGRAMA 3 - FACTORIAL
    cout << "Tercer Programa: FACTORIAL" << endl;
    programa3 p3;
    cout << "Programa:" << endl;
    p3.listar();
    cout << endl << "Ejecucion: " << endl;
    p3.ejecutar();

}

File: 2024-2025/tecnologia_programacion/procesador_instrucciones/java/Main.java

public class Main
{
    public static void main(String[] args)
    {
        // PRUEBA PROGRAMA 1 - SUMA DE DOS NÚMEROS
        System.out.println("Primer programa: SUMA DE DOS NUMEROS");
        Programa1 p1 = new Programa1();
        System.out.println("Programa:");
        p1.listar();
        System.out.println();
        System.out.println("Ejecución:");
        p1.ejecutar();
        System.out.println();

        // PRUEBA PROGRAMA 2 - CUENTA ATRÁS
        System.out.println("Segundo programa: CUENTA ATRAS");
        Programa2 p2 = new Programa2();
        System.out.println("Programa:");
        p2.listar();
        System.out.println();
        System.out.println("Ejecución");
        p2.ejecutar();
        System.out.println();

        // PRUEBA PROGRAMA 3 - FACTORIAL
        System.out.println("Tercer programa: FACTORIAL");
        Programa3 p3 = new Programa3();
        System.out.println("Programa:");
        p3.listar();
        System.out.println();
        System.out.println("Ejecución");
        p3.ejecutar();
    }
}